src/string_immutable.zig

```diff
@@ -3578,6 +3578,48 @@ pub fn NewLengthSorter(comptime Type: type, comptime field: string) type {
     };
 }
 
+pub fn NewGlobLengthSorter(comptime Type: type, comptime field: string) type {
+    return struct {
+        const GlobLengthSorter = @This();
+        pub fn lessThan(_: GlobLengthSorter, lhs: Type, rhs: Type) bool {
+            // Assert: keyA ends with "/" or contains only a single "*".
+            // Assert: keyB ends with "/" or contains only a single "*".
+            const key_a = @field(lhs, field);
+            const key_b = @field(rhs, field);
+
+            // Let baseLengthA be the index of "*" in keyA plus one, if keyA contains "*", or the length of keyA otherwise.
+            // Let baseLengthB be the index of "*" in keyB plus one, if keyB contains "*", or the length of keyB otherwise.
+            const star_a = indexOfChar(key_a, '*');
+            const star_b = indexOfChar(key_b, '*');
+            const base_length_a = if (star_a) |st_a| st_a else key_a.len;
+            const base_length_b = if (star_b) |st_b| st_b else key_b.len;
+
+            // If baseLengthA is greater than baseLengthB, return -1.
+            // If baseLengthB is greater than baseLengthA, return 1.
+            if (base_length_a > base_length_b)
+                return true;
+            if (base_length_b > base_length_a)
+                return false;
+
+            // If keyA does not contain "*", return 1.
+            // If keyB does not contain "*", return -1.
+            if (star_a == null)
+                return false;
+            if (star_b == null)
+                return true;
+
+            // If the length of keyA is greater than the length of keyB, return -1.
+            // If the length of keyB is greater than the length of keyA, return 1.
+            if (key_a.len > key_b.len)
+                return true;
+            if (key_b.len > key_a.len)
+                return false;
+
+            return false;
```

Comment:
If there are multiple keys with the same value, do we infinite loop?

Reply:
You mean if `std.sort.sort()` would itself infinite loop in that situation? I wouldn't expect so, but I can test that case.